app/adapters/digitalocean/adapter.go

```diff
@@ -33,24 +33,41 @@ func (a *adapter) Token() (*oauth2.Token, error) {
 	return token, nil
 }
 
-func (a *adapter) newRequestFromDomainRule(rule domain.Rule) *godo.FirewallRulesRequest {
-	doSource := &godo.Sources{Addresses: []string{rule.IPNet.String()}}
+func (a *adapter) ToString() string {
```

Comment:
Can't we use Struct embedding for this? We have a factory that should be able to take care of setting the names of the adapter. Let's look into this 🤔

Reply:
It would work if we wanted to extend functionalities on a struct. On a second thought I don't see an easier way right now. Setting it as a value and reading the value would expose the adapter name for mutation and different formatting.